spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/ColumnVectorBuilder.java

```diff
@@ -38,8 +39,10 @@ public ColumnVector build(VectorHolder holder, int numRows) {
       if (holder instanceof VectorHolder.DeletedVectorHolder) {
         return new DeletedColumnVector(Types.BooleanType.get(), isDeleted);
       } else if (holder instanceof ConstantVectorHolder) {
-        return new ConstantColumnVector(
-            Types.IntegerType.get(), numRows, ((ConstantVectorHolder<?>) holder).getConstant());
```

Comment:
This is the primary problem we had: we always assumed metadata columns are integers.